c7n/resources/sagemaker.py

```diff
@@ -107,6 +107,55 @@ def _augment(j):
             return list(filter(None, w.map(_augment, jobs)))
 
 
+@resources.register('sagemaker-transform-job')
+class SagemakerTransformJob(QueryResourceManager):
+
+    class resource_type(object):
+        type = 'None'
+        service = 'sagemaker'
+        enum_spec = ('list_transform_jobs', 'TransformJobSummaries', None)
+        detail_spec = (
+            'describe_transform_job', 'TransformJobName', 'TransformJobName', None)
+        id = 'TransformJobArn'
+        name = 'TransformJobName'
+        date = 'CreationTime'
+        dimension = None
+        filter_name = 'TransformJobArn'
+
+    permissions = (
+        'sagemaker:ListTransformJobs', 'sagemaker:DescribeTransformJobs',
+        'sagemaker:ListTags')
+
+    def __init__(self, ctx, data):
+        super(SagemakerTransformJob, self).__init__(ctx, data)
+        self.queries = QueryFilter.parse(
+            self.data.get('query', [
+                {'StatusEquals': 'InProgress'}]))
+
+    def resources(self, query=None):
+        for q in self.queries:
+            if q is None:
+                continue
+            query = query or {}
+            for k, v in q.items():
+                query[k] = v
+        return super(SagemakerTransformJob, self).resources(query=query)
+
+    def augment(self, jobs):
+        client = local_session(self.session_factory).client('sagemaker')
+
+        def _augment(j):
+            tags = self.retry(client.list_tags,
+                ResourceArn=j['TransformJobArn'])['Tags']
+            j['Tags'] = tags
+            return j
+
+        jobs = super(SagemakerTransformJob, self).augment(jobs)
+        for j in jobs:
+            j['Tags'] = _augment(j)['Tags']
```

Comment:
the assignment is extraneous the augment is already mutating the dictionary.

Reply:
I removed the reassignment. wrt to previous PR added a filter_name, maintaining the id and name, and switched to looping through jobs in augment.